### PR TITLE
[onkyo] Add volume scaling variants

### PR DESCRIPTION
**Pull request by cguedel**
This adds a configuration parameter to the onkyo receiver thing that allows to control volume scaling. The following 4 variants are supported:

- 0-100 -> no scaling
- 0-200 -> 0.5 increments for volume, scale by 2
- 0-80  -> volume goes from 0-80, scale by 0.8
- 0-50  -> volume goes from 0-50, scale by 0.5

The default behavior remains unchanged. Volume limiting is applied before the scaling, so should still be ok.

### CODE REVIEW

**Comment by cguedel** [comment]
Note: Now that I think about it, it would of course also be possible to expose the scaling parameter directly. Let me know if this would be preferred, I'm happy to adjust the PR with this functionality.

File: addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml [hunk 1]
```diff
@@ -29,10 +29,16 @@
 		</parameter>
 		<parameter name="volumeLimit" type="integer" min="0" max="100">
 			<label>Volume Limit</label>
-			<description>Limit maximum volume level to defined value.</description>
+			<description>Limit maximum volume level to defined percentage.</description>
 			<default>100</default>
 			<advanced>true</advanced>
 		</parameter>
+		<parameter name="volumeScale" type="integer" min="0" max="100">
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml]: Please apply the formatter in this file.

File: addons/binding/org.openhab.binding.onkyo/README.md [hunk 2]
```diff
@@ -90,6 +90,24 @@ onkyo:onkyoAVR:avr-livingroom [ipAddress="192.168.1.100", port=60128, volumeLimi
 
 Binding then automatically scale the volume level in both directions (100% = 50 = 100%).
 
+You can also change the way volume scaling works. This can be necessary if your receiver uses a different scaling system than 0-100.
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onkyo/README.md]: Can you add a new line after a sentence. It provides better maintainability.

File: addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml [hunk 3]
```diff
@@ -29,10 +29,16 @@
...
+            <label>Volume Scale</label>
+            <description>Select volume scale (0 = 0-100, 1 = 0-100 in 0.5 steps, 2 = 0-80, 3 = 0-50)</description>
+            <default>0</default>
+            <advanced>true</advanced>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml]: Maybe add a fixed options list here to make sure the parameter only accepts one of the four expected values.

File: addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml (hunk 1 above)
Comment (cguedel) [reply addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml]: Done

File: addons/binding/org.openhab.binding.onkyo/README.md (hunk 2 above)
Comment (cguedel) [reply addons/binding/org.openhab.binding.onkyo/README.md]: Done

File: addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml (hunk 3 above)
Comment (cguedel) [reply addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml]: Done -> coefficient is now exposed directly as this provides more flexibility

File: addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml [hunk 4]
```diff
@@ -29,10 +30,16 @@
...
 			<label>Volume Limit</label>
-			<description>Limit maximum volume level to defined value.</description>
+			<description>Limit maximum volume level to defined percentage.</description>
 			<default>100</default>
 			<advanced>true</advanced>
 		</parameter>
+		<parameter name="volumeScale" type="decimal">
+			<label>Volume Scale</label>
+			<description>Applies a scale to the volume.</description>
+			<default>1</default>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml]: What about adding this:
```suggestion
			<options>
				<option value="1">0-100</option>
				<option value="2">0-100 in 0.5 steps</option>
				<option value="0.8">0-80</option>
				<option value="0.5">0-50</option>
			</options>
			<default>1</default>
```

Comment (cguedel) [reply addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml]: Neat, didn't know that was possible. Added this 👍

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.onkyo/ESH-INF/config/config.xml]: Nice. With this you will have a dropdown in some UIs (e.g. Paper UI) to select one of the four possible values.

**Comment by cguedel** [comment]
@cweitkamp Forgot about Zone 2 + Zone 3 volume handling, but changed that now too.

**Pull request merged**